apm-agent-plugins/apm-urlconnection-plugin/src/main/java/co/elastic/apm/agent/urlconnection/HttpUrlConnectionInstrumentation.java

```diff
@@ -51,8 +50,15 @@
 
 public abstract class HttpUrlConnectionInstrumentation extends TracerAwareInstrumentation {
 
-    @VisibleForAdvice
-    public static final WeakConcurrentMap<HttpURLConnection, Span> inFlightSpans = WeakMapSupplier.createMap();
+    private static final WeakConcurrentMap<HttpURLConnection, Span> inFlightSpans = WeakMapSupplier.createMap();
+
+    // Used instead of inspecting sun.net.www.protocol.http.HttpURLConnection.connecting which was added in Java 8
+    private static final ThreadLocal<Boolean> connecting = new ThreadLocal<>() {
```

Comment:
Creating a subclass of `ThreadLocal` causes leaks. See also https://cwiki.apache.org/confluence/display/TOMCAT/MemoryLeakProtection#MemoryLeakProtection-customThreadLocal.

Either use no initial value or use `CallDepth` which seems to be applicable to this case.

Reply:
Ahh, the anonymous class...
Thanks!